pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -23,10 +38,10 @@ def register_slurm_signal_handlers(self):
 
         if on_slurm:
             log.info("Set SLURM handle signals.")
-            signal.signal(signal.SIGUSR1, self.sig_handler)
+            signal.signal(signal.SIGUSR1, self.sig_slurm_handler)
             signal.signal(signal.SIGTERM, self.term_handler)
 
-    def sig_handler(self, signum, frame):  # pragma: no-cover
+    def sig_slurm_handler(self, signum, frame):  # pragma: no-cover
```

Comment:
I would instead much prefer:
- The checkpoint saving for fault tolerance and the slurm signal handler use the same code
- After the fault-tolerant checkpoint is saved, allow the SLURM cluster environment to dictate whether job resubmission should happen.

otherwise there are no guarantees `hpc_save` keeps up : https://github.com/PyTorchLightning/pytorch-lightning/issues/9407

Reply:
Hey @ananthsub. We don't have access to a SLURM cluster, therefore I don't want to touch this feature as it is quite simple to break and lot of users are relying on this.

And yes, hpc_save would be to be refactored in further PRs.